appsettings load in fail fix

niraymak opened the pull request:
## Description

Renamed appsettings to fix an issue where API loaded the wrong appsettings in docker environment

All appsettings now have unique names.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Run project using docker-compose up.
2. Check if all projects are running

**Please check if notification server starts. I had problems today, but earlier aswell, so it might be my machine.**

## Link to issue

Closes: #349

Brend-Smits commented:
My notification service does not want to start (even after Rabbit MQ is confirmed running/working).
https://gist.github.com/Brend-Smits/64591ba4d45de00f32cb580273fc447f

Brend-Smits commented:
I would also consider this to be a workaround and not an actual fix ^^', could you perhaps comment in the code above the appsettings init code why they have unique names?

niraymak commented:
Changed the variables to come hardcoded from docker-compose.yml for local docker usage. A few users made a custom .env file but I think this is an extra step which is unnecessary. Also added comments in appsettings.json files to make clear why the files are renamed.

Brend-Smits commented on API/appsettingsapi.Development.json:
```diff
@@ -1,7 +1,8 @@
+// Appsettings is renamed because of an issue where the project loaded another appsettings.json
```
No comments in JSON please (not allowed)

Brend-Smits commented on API/appsettingsapi.json:
```diff
@@ -1,3 +1,5 @@
+// Appsettings is renamed because of an issue where the project loaded another appsettings.json
```
No comments in JSON please (not allowed)

Brend-Smits commented on NotificationSystem/appsettingsnotificationsystem.Development.json:
```diff
@@ -1,3 +1,5 @@
+// Appsettings is renamed because of an issue where the project loaded another appsettings.json
```
No comments in JSON please (not allowed)

Brend-Smits commented on NotificationSystem/appsettingsnotificationsystem.json:
```diff
@@ -1,3 +1,5 @@
+// Appsettings is renamed because of an issue where the project loaded another appsettings.json
```
No comments in JSON please (not allowed)

Brend-Smits commented on docker-compose.yml:
```diff
@@ -5,8 +5,8 @@ services:
     environment:
-      - RABBITMQ_USERNAME=${App__RabbitMQ__Username}
-      - RABBITMQ_PASSWORD=${App__RabbitMQ__Password}
+      - RABBITMQ_USERNAME=test
```
This is purely for development docker-compose right?

niraymak replied:
Correct. :)

Brend-Smits reviewed (approved):
Nice!
